amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseTaker.java

```diff
@@ -230,6 +239,29 @@ public DynamoDBLeaseTaker withMaxLeasesToStealAtOneTime(int maxLeasesToStealAtOn
         return takenLeases;
     }
 
+    /**
+     * If update all leases takes longer than the lease renewal time,
+     * we fetch the latest lease info for the given lease. If nothing is found (or any transient network error occurs),
+     * we default to the last known state of the lease
+     *
+     * @param updateAllLeasesEndTime How long it takes for update all leases to complete
+     * @return set of leases to take.
+     */
+    private Set<Lease> getCurrentLeaseStateForExceededLeaseUpdateTime(long updateAllLeasesEndTime,
+                                                                      Set<Lease> leasesToTake) {
+        if (updateAllLeasesEndTime > leaseRenewalIntervalMillis) {
```

Comment:
We might need to add some slack time here. May be if time taken for first update is > 0.5 * leaseRenewalIntervalMillis, then we can renew it again. Can we collect this data point for big, medium and small regions?